test/batcher.js

```diff
@@ -197,88 +205,88 @@ contract("Batcher", (accounts) => {
         );
       } else {
         assertBalanceDiff(
           startBalance,
           endBalance,
           value,
-          "Valid account didn't receive funds"
+          'Valid account didn\'t receive funds'
         );
       }
     }
+
+    //Check sender balances
     const testBatcherBalance = await getBalance(
       testBatcherDriverInstance.address
     );
     assert.strictEqual(
-      testBatcherBalance,
+      parseInt(testBatcherBalance),
       expectedRetVal,
-      "TestBatcherDriver has more or less funds left over than it should"
+      'Sender should not receive funds unless it is a recipient.'
     );
-  };
 
-  afterEach(async () => {
+    //Check batcher contract balances
     assert.strictEqual(
-      await getBalance(batcherInstance.address),
-      "0",
-      "Batcher balance should be zero after transaction"
+      parseInt(await getBalance(batcherInstance.address)),
+      extraValue,
+      'Batcher balance is not equal to expected value.'
     );
-  });
+  };
 
-  describe("Good transactions single recipient", () => {
-    it("Sends batch with one recipient", async () => {
+  describe('Good transactions single recipient', () => {
+    it('Sends batch with one recipient', async () => {
       const params = {
         recipients: [accounts[1]],
         values: [5]
       };
       await runTestBatcherDriver(params);
     });
 
-    it("Deflects reentry and returns amount to sender", async () => {
+    it('Deflects reentry and returns amount to sender', async () => {
       const params = {
         recipients: [reentryInstance.address],
         values: [5],
         extraValue: 5,
-        expectedRetVal: "10",
+        expectedRetVal: 10,
         expectOverallFailure: true,
         expectedErrMsg: sendFailedErrorMsg
       };
       await runTestBatcherDriver(params);
     });
 
-    it("Returns amount to sender when a recipient fails", async () => {
+    it('Returns amount to sender when a recipient fails', async () => {
       const params = {
         recipients: [failInstance.address],
         values: [5],
-        expectedRetVal: "5",
+        expectedRetVal: 5,
         expectOverallFailure: true,
         expectedErrMsg: sendFailedErrorMsg
       };
       await runTestBatcherDriver(params);
     });
   });
 
-  describe("Good transactions multiple recipients", () => {
-    it("Correctly sends with exact amount", async () => {
+  describe('Good transactions multiple recipients', () => {
+    it('Correctly sends with exact amount', async () => {
       const params = {
         recipients: accounts.slice(1, 4),
         values: createRandIntArr(3)
       };
       await runTestBatcherDriver(params);
     });
 
-    it("Correctly sends with extra value", async () => {
+    it('Correctly sends with extra value', async () => {
       const params = {
         recipients: accounts.slice(1, 4),
-        values: createRandIntArr(3),
+        values: [5, 5, 10],
```

Comment:
Why are you removing the randomness here? Randomness allows for some edge cases to be [potentially] captured automatically